### PR TITLE
Implement 'range' property for hover response

### PR DESCRIPTION
**Pull request by vhakulinen**
Include `range` property for the `textDocument/hover` response: https://microsoft.github.io/language-server-protocol/specification#textDocument_hover.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/python-language-server](https://github.com/palantir/python-language-server), @vhakulinen! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: pyls/plugins/hover.py [hunk 1]
```diff
@@ -17,4 +18,9 @@ def pyls_hover(document, position):
         # :(
         return {'contents': ''}
 
-    return {'contents': _utils.format_docstring(definitions[0].docstring()) or ""}
+    pos = document.word_range_at_position(position)
+
+    return {
+        'contents': _utils.format_docstring(definitions[0].docstring()) or "",
+        'range': pos,
```
Comment (gatesn) [inline pyls/plugins/hover.py:25]: Can you inline `pos` here

File: pyls/plugins/hover.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Copyright 2017 Palantir Technologies, Inc.
 import logging
+import json
```
Comment (gatesn) [inline pyls/plugins/hover.py:3]: Unused?

File: pyls/workspace.py [hunk 3]
```diff
@@ -178,6 +178,28 @@ def offset_at_position(self, position):
         """Return the byte-offset pointed at by the given position."""
         return position['character'] + len(''.join(self.lines[:position['line']]))
 
+    def word_range_at_position(self, position):
+        """Get the range of a word at given position."""
+        if position['line'] >= len(self.lines):
+            return None
+
+        line = self.lines[position['line']]
```
Comment (gatesn) [inline pyls/workspace.py:186]: Can you add a test for this method explicitly?

**Review by gatesn — CHANGES_REQUESTED** [review]
Thank you for this contribution, really nice. Couple of small comments then should be good to go

**Comment by gatesn** [comment]
Copying over to a local branch so I can finish up the changes

**Comment by vhakulinen** [comment]
Sorry, I completely forgot this while doing other stuff.

**Pull request merged**